components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -2049,6 +2052,48 @@ private String interpretQueryString(AuthenticationContext context, String queryS
         return queryString;
     }
 
+    /**
+     * Gets the value of the parameter corresponding to the given parameter
+     * name from the request's parameter map.
+     *
+     * @param parameters    The parameter map of the request.
+     * @param parameterName The name of the parameter to be retrieved.
+     * @return The value of the parameter if it is present in the parameter map.
+     * If it is not present, an empty String is returned instead.
+     */
+    private String getParameterFromParamMap(Map<String, String[]> parameters, String parameterName) {
+
+        String[] parameterValueMap = parameters.get(parameterName);
+        if (parameterValueMap != null && parameterValueMap.length > 0) {
+            return parameterValueMap[0];
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * Parses the given URI String to get the parameter value corresponding to the
+     * given parameter name.
+     *
+     * @param uriString     The URI String to be parsed.
+     * @param parameterName The name of the parameter to be retrieved.
+     * @return The value of the parameter if it is present in the URI String.
+     * If it is not present, an empty String is returned instead.
+     */
+    private String getParameterFromURIString(String uriString, String parameterName) {
+
+        if (StringUtils.isNotBlank(uriString)) {
+            String[] queryParams = uriString.split(URI_QUERY_PARAM_DELIMITER, -1);
```

Comment:
Is there any specific reason to use -1 as the limit to the split function here?

Reply:
If the limit is negative then the pattern will be applied as many times as possible rather than a fixed number of times, and we validate the length of the split components during processing.